clang/lib/Sema/SemaExpr.cpp

```diff
@@ -62,7 +63,42 @@
 
 using namespace clang;
 using namespace sema;
+class GetSymbolicName {
+private:  
+std::string LastString;
+public:
+
+    void VisitDeclRefExpr(DeclRefExpr *Node) {
+       if (const auto *ECD = dyn_cast<EnumConstantDecl>(Node->getDecl())) {
+                if (const auto *ED = dyn_cast<EnumDecl>(ECD->getDeclContext())) {
+                    // Construct the fully qualified name
+                    LastString = ED->getName().str() + "::" + ECD->getName().str();
+                }
+            } else {
+                // Handle other declarations (e.g., variables, functions)
+                LastString = Node->getDecl()->getName().str();
+            }
```

Comment:
Shouldn't you throw an error if it's not an Enum decl?
I assume this class is there to fetch the name of Enum decl given inside the `nameof` operator. In that case, shouldn't you issue an error if you find anything other than an Enum?